doc/nrf/libraries/modem/location.rst

```diff
@@ -0,0 +1,198 @@
+* :kconfig:`CONFIG_NRF_CLOUD_AGPS` - Enables A-GPS data retrieval from `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_PGPS` - Enables P-GPS data retrieval from `nRF Cloud`_.
+
+The following options control the transport used with `nRF Cloud`_:
+
+* :kconfig:`CONFIG_NRF_CLOUD_REST` - Uses REST APIs to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_NRF_CLOUD_MQTT` - Uses MQTT transport to communicate with `nRF Cloud`_.
+* :kconfig:`CONFIG_REST_CLIENT` - Enable :ref:`lib_rest_client` library.
+
+Both cellular and WiFi location services are selected utilizing runtime configuration but the available services must be configured first.
+For cellular location services, use at least one of the following sets of options and configure corresponding authentication parameters (more details and configuration options can be found from :ref:`lib_multicell_location`):
+
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_HERE` and :kconfig:`CONFIG_MULTICELL_LOCATION_HERE_API_KEY` (see below other authentication options)
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_SKYHOOK` and :kconfig:`CONFIG_MULTICELL_LOCATION_SKYHOOK_API_KEY`
+* :kconfig:`CONFIG_MULTICELL_LOCATION_SERVICE_POLTE` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_CUSTOMER_ID` and :kconfig:`CONFIG_MULTICELL_LOCATION_POLTE_API_TOKEN`
+
+For WiFi location services, use at least one of the following sets of options and configure corresponding authentication parameters:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_NRF_CLOUD`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_API_KEY`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK` and :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_API_KEY`
+
+Following WiFi service related options can usually have default values:
+
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_HERE_TLS_SEC_TAG`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_HOSTNAME`
+* :kconfig:`CONFIG_LOCATION_METHOD_WIFI_SERVICE_SKYHOOK_TLS_SEC_TAG`
+
+Usage
+*****
+
+Using Location library is rather easy. First you need to initialize the library with :c:func:`location_init`.
+
+Secondly, you need to set configuration (:c:struct:`loc_config`) including location method configurations (:c:struct:`loc_method_config`).
+This is achieved easily by setting first default values by using :c:func:`loc_config_defaults_set` and c:func:`loc_config_method_defaults_set`,
+and then setting any required non-default values to the structures.
+
+Once the configuration is set up properly, you just need to call c:func:`location_request` with the configuration.
+
+Samples using the library
+*************************
+
+The following |NCS| samples use this library:
+
+* :ref:`location_sample`
```

Comment:
@peknis I guess I need your help here. This reference causes document build to fail:
`/home/runner/work/sdk-nrf/sdk-nrf/ncs/nrf/doc/_build/nrf/src/libraries/modem/location.rst:157:Failed to create a cross reference. A title or caption not found: location_sample`

README.rst of Location sample has this line in the beginning:
`.. _location_sample:`

Is the problem that these are both new files and the build is not able to find it because of that? I hope this is a typo that I still cannot see...

Reply:
This is fixed now by moving the |sample path| to correct place. It broke the link. Thanks Pekka!